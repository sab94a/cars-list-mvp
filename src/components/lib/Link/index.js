// @flow

import React from 'react';
import type { Node, AbstractComponent } from 'react';
import type { Path } from 'components/lib/types';

type Props = {
    children: Node,
    href: Path
}

class Link extends React.Component<Props> {
    static Component: AbstractComponent<*> | string = 'a';

    render() {
        const { href, children, ...rest } = this.props;

        return (
            <Link.Component { ...rest } href={ href }>{ children }</Link.Component>
        );
    }
}

export default Link;
