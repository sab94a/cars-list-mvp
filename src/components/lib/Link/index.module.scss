@import 'components/lib/variables.scss';

.root {
    color: $primary;

    &:hover, &:focus {
        text-decoration: underline;
        color: $primary-pressed;    
    }
}
