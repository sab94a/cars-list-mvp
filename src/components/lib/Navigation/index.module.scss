@import 'components/lib/variables.scss';

.root {
    display: flex;
    flex-direction: row;
    height: $panel-height;
}

.link {
    display: flex;
    align-items: center;
    font-size: $font-size-sm;
    padding: 0 $offset-lg;
    text-decoration: none;
}