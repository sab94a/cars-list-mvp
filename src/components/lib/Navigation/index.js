// @flow

import React from 'react';
import Link from  'components/lib/Link';
import type { NavItem } from 'components/lib/types';
import styles from './index.module.scss';

type Props = {
    items: Array<NavItem>
};

const Navigation = ({ items }: Props) => (
    <nav className={ styles.root }>
        { items.map(({ href, title }:NavItem, index:number) => (
            <Link key={ index } href={ href } className={ styles.link }>{ title }</Link>
        )) }
    </nav>
);

export default React.memo<Props>(Navigation);
