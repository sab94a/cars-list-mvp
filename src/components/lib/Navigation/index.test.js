import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Link from 'components/lib/Link';
import Navigation from './';

const navigation = [{
    href: 'href',
    title: 'title'
}, {
    href: 'href2',
    title: 'title2'
}]

describe('<Navigation />', () => {
    let component = null
    
    beforeEach(() => {
        component = mount(<Navigation items={ navigation } />);
    });

    it('Should be rendered correctly', () => {
        expect(toJson(component)).toMatchSnapshot();
    });

    it('Should render right amount of Link', () => {
        expect(component.find(Link)).toHaveLength(2);
    });
});
