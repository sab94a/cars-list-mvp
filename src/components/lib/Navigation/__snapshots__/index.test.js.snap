// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> Should be rendered correctly 1`] = `
<Memo(Navigation)
  items={
    Array [
      Object {
        "href": "href",
        "title": "title",
      },
      Object {
        "href": "href2",
        "title": "title2",
      },
    ]
  }
>
  <nav
    className="root"
  >
    <Link
      className="link"
      href="href"
      key="0"
    >
      <a
        className="link"
        href="href"
      >
        title
      </a>
    </Link>
    <Link
      className="link"
      href="href2"
      key="1"
    >
      <a
        className="link"
        href="href2"
      >
        title2
      </a>
    </Link>
  </nav>
</Memo(Navigation)>
`;
